MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/ProductServiceDefault.java

```diff
@@ -121,14 +144,29 @@ public CancellableTask getProduct(final Long productId, final Callback<Product>
             throw new NullPointerException("productId cannot be null");
         }
 
-        return retrofitService
+        final Observable<Product> apiRequest = retrofitService
             .getProducts(appId, String.valueOf(productId))
             .retryWhen(networkRetryPolicyProvider.provide())
             .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>())
             .compose(new UnwrapRetrofitBodyTransformer<ProductListings, List<Product>>())
             .compose(new FirstListItemOrDefaultTransformer<Product>())
-            .onErrorResumeNext(new BuyClientExceptionHandler<Product>())
-            .observeOn(callbackScheduler);
+            .onErrorResumeNext(new BuyClientExceptionHandler<Product>());
+
+        return ApiRequestInterceptWrapper.wrap(
+            apiRequest,
+            new ApiRequestInterceptWrapper.InterceptorCall<ProductApiInterceptor, Product>(requestInterceptor) {
```

Comment:
something is off here.  Both the request and response do exactly the same thing - can we factor this out?